### PR TITLE
Initalize

### PR DESCRIPTION
**Pull request by chris-pardy**
Setup the Redux-compose package

### CODE REVIEW

**Comment by chris-pardy** [comment]
@kstrickland0612 having not really used many of our Redux helpers before I was hoping you especially could take a look at the Readme and see if it made sense.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,214 @@
...
+export const bothReducer = ofType(
+  ['Increment', ({ type }) => type === 'Decrement'],
+  (state, action) => incrementReducer(decrementReducer(state, action), action)
+);
+```
+
+### `withDefault(defaultState, reducer?)`
+
+Given a default state and optionally a reducer, returns a reducer which
+will use the default state when undefined is passed. If the second argument
```
Comment (kstrickland0612) [inline README.md]: Is this saying the returned reducer will use default state (first argument, which is { value: 2 } in the example below) when action.type is undefined? Trying to make sure I understand where the undefined is coming from.

Also small typo - provided is misspelled as "previded" in this paragraph.

**Comment by chris-pardy** [comment]
@kstrickland0612 updated the documentation to hopefully make it clearer.
Also I added TSDoc into everything so intellisense works better.

@modalton @qcharlieshi @brentshields @kisaraofpern 
This is very much what we have no in Internal-Dashboard with a few addendums.
However before we cut this as version 1 of a public library we could go in a slightly different direction.
```ts
// some generic function combinators
const reducer = composeReducers(
   filter(predicate, firstReducer),
   focus(lens, secondReducer)
);
```

```ts
// some helper functions to create the predicates and lenses
const typedPredicate = ofType('INCREMENT');
const pathLens = forPath(['foo','bar']);
const defaultValueLens = withDefault({ foo: { bar: 0 }});
```

The advantage of going this way is that each of the functions would be really simple
to use as the basis for lots of different reducer setups.

**Comment by chris-pardy** [comment]
@kstrickland0612 updated the documentation to hopefully make it clearer.
Also I added TSDoc into everything so intellisense works better.

**Pull request merged**